From shimoda.yoshihiro@renesas.com  Mon Oct  8 14:07:00 2007
From: Yoshihiro Shimoda <shimoda.yoshihiro@renesas.com>
Date: Fri, 05 Oct 2007 15:53:12 +0900
Subject: USB: r8a66597-hcd: fix driver removing
To: greg@kroah.com
Cc: linux-usb-devel@lists.sourceforge.net
Message-ID: <4705DF58.2080204@renesas.com>


Fixed the problem that accessed register of this controller after
having called iounmap().

Signed-off-by: Yoshihiro Shimoda <shimoda.yoshihiro@renesas.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/host/r8a66597-hcd.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/usb/host/r8a66597-hcd.c
+++ b/drivers/usb/host/r8a66597-hcd.c
@@ -2126,8 +2126,8 @@ static int __init_or_module r8a66597_rem
 	struct usb_hcd		*hcd = r8a66597_to_hcd(r8a66597);
 
 	del_timer_sync(&r8a66597->rh_timer);
-	iounmap((void *)r8a66597->reg);
 	usb_remove_hcd(hcd);
+	iounmap((void *)r8a66597->reg);
 	usb_put_hcd(hcd);
 	return 0;
 }
